### PR TITLE
add BuddyConvolution

### PR DESCRIPTION
**Pull request by FloatingcloudKnight**
add conv2d examples and makefile target

### CODE REVIEW

File: examples/BuddyConvolution/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+log.mlir
+log.ll
+log.s
```
Comment (zhanghb97) [inline examples/BuddyConvolution/.gitignore:3]: Please add an empty line here.

File: examples/BuddyConvolution/makefile [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		-arith-bufferize \
+		-convert-scf-to-cf \
+		-convert-vector-to-llvm \
+		-convert-arith-to-llvm \
+		-finalize-memref-to-llvm \
+		-llvm-request-c-wrappers \
+		-convert-func-to-llvm \
+		-reconcile-unrealized-casts | \
+	${MLIR_CPU_RUNNER} ${OPT_FLAG} -e main -entry-point-result=void \
+		-shared-libs=${MLIR_RUNNER_UTILS} -shared-libs=${MLIR_C_RUNNER_UTILS}
```
Comment (zhanghb97) [inline examples/BuddyConvolution/makefile:69]: Please add an empty line here.

File: examples/BuddyConvolution/conv2d.mlir [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    %print_v2 = memref.cast %v2 : memref<?x?xf32> to memref<*xf32>
+    call @printMemrefF32(%print_v2) : (memref<*xf32>) -> ()
+
+    memref.dealloc %v0 : memref<?x?xf32>
+    memref.dealloc %v1 : memref<?x?xf32>
+    memref.dealloc %v2 : memref<?x?xf32>
+    return
+  }
+}
```
Comment (zhanghb97) [inline examples/BuddyConvolution/conv2d.mlir:54]: Please add an empty line here.

File: examples/BuddyConvolution/CMakeLists.txt
Comment (zhanghb97) [inline examples/BuddyConvolution/CMakeLists.txt:1]: Please remove this file.

File: examples/BuddyConvolution/makefile [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+MLIR_RUNNER_UTILS := ../../llvm/build/lib/libmlir_runner_utils.so
+MLIR_C_RUNNER_UTILS := ../../llvm/build/lib/libmlir_c_runner_utils.so
+MTRIPLE := x86_64-unknown-linux-gnu
+else ifeq ($(shell uname),Darwin)
+MLIR_RUNNER_UTILS := ../../llvm/build/lib/libmlir_runner_utils.dylib
+MLIR_C_RUNNER_UTILS := ../../llvm/build/lib/libmlir_c_runner_utils.dylib
+MTRIPLE := x86_64-apple-darwin
+endif
+
+test-lower:
```
Comment (zhanghb97) [inline examples/BuddyConvolution/makefile:19]: Please remove this target.

File: examples/BuddyConvolution/conv2d.mlir
Comment (zhanghb97) [inline examples/BuddyConvolution/conv2d.mlir:1]: Please add the test.
If you run the `ninja check-buddy`, you will get the failed report.
See [here](https://github.com/buddy-compiler/buddy-mlir/commit/768f1f7722c36608cd036b293c95ad1c2fa2d5d0#diff-fcc233b7c6e44b034e04750a4a53afa906effad5c30d6a41704075ba73bd32cf) as an example.

**Review by zhanghb97 — CHANGES_REQUESTED** [review]

**Review by FloatingcloudKnight — COMMENTED** [review]
update BuddyConvolution

**Review by zhanghb97 — APPROVED** [review]
Good first step! Thanks!

**Pull request merged**